modelkit/assets/drivers/s3.py

```diff
@@ -1,45 +1,49 @@
 import os
+from typing import Any, Optional
 
 import boto3
 import botocore
-import pydantic
-from pydantic import BaseSettings
 from structlog import get_logger
 from tenacity import retry
 
 from modelkit.assets import errors
+from modelkit.assets.drivers.abc import StorageDriver
 from modelkit.assets.drivers.retry import RETRY_POLICY
 
 logger = get_logger(__name__)
 
 
-class S3DriverSettings(BaseSettings):
-    bucket: str = pydantic.Field(..., env="MODELKIT_STORAGE_BUCKET")
-    aws_access_key_id: str = pydantic.Field(None, env="AWS_ACCESS_KEY_ID")
-    aws_secret_access_key: str = pydantic.Field(None, env="AWS_SECRET_ACCESS_KEY")
-    aws_default_region: str = pydantic.Field(None, env="AWS_DEFAULT_REGION")
-    aws_session_token: str = pydantic.Field(None, env="AWS_SESSION_TOKEN")
-    s3_endpoint: str = pydantic.Field(None, env="S3_ENDPOINT")
+class S3StorageDriver(StorageDriver):
+    bucket: str
+    endpoint_url: Optional[str]
+    client: Any
 
-    class Config:
-        extra = "forbid"
+    def __init__(
+        self,
+        bucket: str = None,
+        aws_access_key_id: str = None,
+        aws_secret_access_key: str = None,
+        aws_default_region: str = None,
+        aws_session_token: str = None,
+        s3_endpoint: str = None,
+    ):
 
-
-class S3StorageDriver:
-    def __init__(self, settings: S3DriverSettings = None):
-        if not settings:
-            settings = S3DriverSettings()
-        self.bucket = settings.bucket
-        self.endpoint_url = settings.s3_endpoint
+        self.bucket = bucket or os.environ.get("MODELKIT_STORAGE_BUCKET", None)
```

Comment:
NB: ```python         self.bucket = bucket or os.environ.get("MODELKIT_STORAGE_BUCKET")```

Reply:
^ same for every `get` :D

Reply:
Thanks, I also fixed it here, and aligned the bucket init to other drivers